pkg/core/config/tenant.go

```diff
@@ -577,15 +671,31 @@ type WelcomeEmailConfiguration struct {
 }
 
 type SSOConfiguration struct {
-	URLPrefix           string                     `json:"url_prefix" yaml:"url_prefix" msg:"url_prefix"`
-	JSSDKCDNURL         string                     `json:"js_sdk_cdn_url" yaml:"js_sdk_cdn_url" msg:"js_sdk_cdn_url"`
-	StateJWTSecret      string                     `json:"state_jwt_secret" yaml:"state_jwt_secret" msg:"state_jwt_secret"`
-	AutoLinkEnabled     bool                       `json:"auto_link_enabled" yaml:"auto_link_enabled" msg:"auto_link_enabled"`
-	AllowedCallbackURLs []string                   `json:"allowed_callback_urls" yaml:"allowed_callback_urls" msg:"allowed_callback_urls"`
-	Providers           []SSOProviderConfiguration `json:"providers" yaml:"providers" msg:"providers"`
+	CustomToken CustomTokenConfiguration `json:"custom_token" yaml:"custom_token" msg:"custom_token"`
+	OAuth       OAuthConfiguration       `json:"oauth" yaml:"oauth" msg:"oauth"`
 }
 
-func (s *SSOConfiguration) APIEndpoint() string {
+type CustomTokenConfiguration struct {
+	Enabled                    bool   `json:"enabled" yaml:"enabled" msg:"enabled"`
+	Issuer                     string `json:"issuer" yaml:"issuer" msg:"issuer"`
+	Audience                   string `json:"audience" yaml:"audience" msg:"audience"`
+	Secret                     string `json:"secret" yaml:"secret" msg:"secret"`
+	OnUserDuplicateAllowMerge  bool   `json:"on_user_duplicate_allow_merge" yaml:"on_user_duplicate_allow_merge" msg:"on_user_duplicate_allow_merge"`
+	OnUserDuplicateAllowCreate bool   `json:"on_user_duplicate_allow_create" yaml:"on_user_duplicate_allow_create" msg:"on_user_duplicate_allow_create"`
+}
+
+type OAuthConfiguration struct {
+	URLPrefix                      string                       `json:"url_prefix" yaml:"url_prefix" msg:"url_prefix"`
+	JSSDKCDNURL                    string                       `json:"js_sdk_cdn_url" yaml:"js_sdk_cdn_url" msg:"js_sdk_cdn_url"`
+	StateJWTSecret                 string                       `json:"state_jwt_secret" yaml:"state_jwt_secret" msg:"state_jwt_secret"`
+	AllowedCallbackURLs            []string                     `json:"allowed_callback_urls" yaml:"allowed_callback_urls" msg:"allowed_callback_urls"`
```

Comment:
I think `AllowedCallbackURLs` should be required; otherwise, `AuthHandler.validateCallbackURL` would allow any callback URL, which would be a major security problem.